src/EditorFeatures/CSharpTest/SplitComment/SplitCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,249 @@
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.SplitComment
+{
+    [UseExportProvider]
+    public class SplitCommentCommandHandlerTests : AbstractSplitCommentCommandHandlerTests
+    {
+        protected override TestWorkspace CreateWorkspace(string markup)
+            => TestWorkspace.CreateCSharp(markup);
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestMissingWithSelection()
+        {
+            TestNotHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //[|Test|] Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestMissingWithMultiSelection()
+        {
+            TestNotHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //[||]Test[||] Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitStartOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //[||]Test Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //
+        //Test Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitStartOfCommentWithLeadingSpace1()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // [||]Test Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //
+        // Test Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitStartOfCommentWithLeadingSpace2()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //[||] Test Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //
+        // Test Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestCommentWithMultipleLeadingSpaces()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //    X[||]Test Comment
```

Comment:
📝 This test case is great, but can we parameterize the input text `[||]`. The expected output text for this should be exactly the same for all of:

* No space between `X` and `Test` (`"[||]"`)
* One space before the caret between `X` and `Test` (`" [||]"`)
* One space after the caret between `X` and `Test` (`"[||] "`)
* Spaces before and after the caret between `X` and `Test` (`" [||] "`)
* A selected space between `X` and `Test` (`"[| |]"`)

Reply:
> A selected space between X and Test ("[| |]")

Currently we disable the feature if there is a non-empty selection.  Do you think we should tweak that behavior?  I'm not exaclty sure what i think should happen there.

Reply:
have done the rest though.

Reply:
Missed that detail. Would be good to have a test showing behavior for whitespace selection. Other additions look great.

Reply:
Will do.  Note: i'm honestly on hte fence in terms of what behavior there should be for selection.  if you feel it's appropriate to insert the comment in that case just lmk so i stop flip flopping :)